### PR TITLE
test: brain batch dispense

### PR DESCRIPTION
**Pull request by mautematico**
I think I finally understood how batchDispense is suposed to work and have writen a couple of (failling) tests for the BillDispense.factory mock.

This is a WIP so I will be pushing new commits and may be even rewriting my old ones

### CODE REVIEW

**Comment by mautematico** [comment]
It turns out that my test cases were wrong, I have now fixed and I'm adding new ones :)

**Comment by mautematico** [comment]
[Write tests for the batch dispenser code](https://trello.com/c/s3gsPQVi/34-write-tests-for-the-batch-dispenser-code)

File: lib/printer/loader.js [hunk 1]
```diff
@@ -28,18 +29,18 @@ function configureAsZebra (printerCfg) {
 
 function autoDetect (printerCfg) {
   // this can be either nippon, zebra or no printer
-  return zebra.checkStatus(printerCfg, 2000)
+  return zebra().checkStatus(printerCfg, 2000)
     .then(() => {
       // only zebras can respond
       configureAsZebra(printerCfg)
-      return zebra
+      return require('./zebra')
```
Comment (joshmh) [inline lib/printer/loader.js]: Why not ``return zebra()``?

File: lib/printer/loader.js [hunk 2]
```diff
@@ -63,11 +64,11 @@ function load () {
 
     // if no model: some gaias were shipped without a config
     if (!model) {
-      return resolve(zebra)
+      return resolve(require('./zebra'))
```
Comment (joshmh) [inline lib/printer/loader.js]: Why not ``zebra()``?

File: lib/printer/loader.js (hunk 1 above)
Comment (mautematico) [reply lib/printer/loader.js]: oops. fixed

**Pull request merged**